src/material/slider/slider-input.ts

```diff
@@ -500,7 +504,9 @@ export class MatSliderThumb implements _MatSliderThumb, OnDestroy, ControlValueA
    * @docs-private
    */
   writeValue(value: any): void {
-    this.value = value;
+    if (this._onChangeFn) {
+      this.value = value;
```

Comment:
It seems a little strange that we don't assigned the value if there's a CVA. Shouldn't the value always be up-to-date?

Reply:
This is specifically to handle NgModel will emit `null` before being initialized https://github.com/angular/angular/issues/14988. The `null` value is only emitted before`registerOnChange` gets called

Reply:
Doesn't the forms module also emit the actual value when it comes in afterwards?

Reply:
Yes but because setting the slider thumb values equal to `0` (`null` gets coerced to `0`) will also update their mins and maxes. This means that when the correct values get emitted the native inputs will ignore the update due to their new (incorrect) min/max constraints

Here is a full example of how this can break initialization:

Slider prop initialization sets default values:
```
Start Thumb: value 0, min 0, max 1
End Thumb:   value 1, min 0, max 1
```
NgModel calls `writeValue` with `null` which is coerced to `0`.

NOTE: The min & max are updated according to each other's values.
```
Start Thumb: value 0, min 0, max 0
End Thumb:   value 0, min 0, max 1
```
NgModel calls `writeValue` with `0.25` for the start value.

NOTE: The start value does not change because it cannot be greater than the end thumb's value.
```
Start Thumb: value 0, min 0, max 0
End Thumb:   value 0, min 0, max 0
```
NgModel calls `writeValue` with `0.75` for the start value.
```
Start Thumb: value 0, min 0, max 0.75
End Thumb:   value 0.75, min 0, max 0
```

Reply:
It's your call, but at it could at least use a comment around why we're null checking the change callback before assigning the value. Alternatively we could have some sort of `isInitialized` flag which should be a bit more obvious.

Reply:
Sounds good. I created a flag and left a comment describing the issue as its jsdoc